Database/Repositories/Implementations/PlayerRatingsRepository.cs

```diff
@@ -24,33 +24,31 @@ public async Task<IEnumerable<PlayerRating>> GetAsync(long osuPlayerId)
         return await _context.PlayerRatings.Where(x => x.PlayerId == id.Value).ToListAsync();
     }
 
-    public async Task<PlayerRating?> GetAsync(int playerId, Ruleset ruleset) =>
-        await _context.PlayerRatings.Where(x => x.PlayerId == playerId && x.Ruleset == ruleset).FirstOrDefaultAsync();
-
-    public async Task<int> BatchInsertAsync(IEnumerable<PlayerRating> playerRatings)
+    public async Task<PlayerRating?> GetAsync(int playerId, Ruleset ruleset)
     {
-        var ls = new List<PlayerRating>();
-        foreach (PlayerRating stat in playerRatings)
+        var results = await _context.PlayerRatings
+            .Include(pr => pr.Adjustments)
+            .Where(pr => pr.PlayerId == playerId && pr.Ruleset == ruleset)
+            .Select(pr => new
+            {
+                PlayerRating =
+                    pr, // TODO: Should probably include all RatingAdjustments regardless of type and have web not display initial
+                Adjustments = pr.Adjustments.Where(ra => ra.AdjustmentType != RatingAdjustmentType.Initial)
+                    .OrderBy(ra => ra.Timestamp)
+                    .ToList()
+            })
```

Comment:
What is with this projection? You can do a filtered include on the adjustments if that is the goal

Reply:
The projection here is to execute a subquery which fetches the `MatchCost`. You can do subqueries by projecting via `Select` but not by `Where`.

Reply:
yeah this was kinda scuffed